File: selfstudy/js/transactions_max.js
/*
task 32:
Transactions - Max
You need to find the maximum value among transaction amounts in dollars
*/

function highest_transactions(amounts) {
  return amounts.length ? Math.max(...amounts) : 0;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: highest_transactions([10, 20, 30]) },
  { input: [50, 60, 70], expected: highest_transactions([50, 60, 70]) },
  { input: [1, 2, 3], expected: highest_transactions([1, 2, 3]) },
  { input: [100, 200], expected: highest_transactions([100, 200]) },
  { input: [], expected: highest_transactions([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('highest_transactions', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(highest_transactions(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, highest_transactions(t.input) === t.expected || JSON.stringify(highest_transactions(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works

File: selfstudy/golang/transactions_max.go
/*
task 232:
Transactions - Max
You need to find the maximum value among transaction amounts in dollars
*/

package main

import (
	"fmt"
	"reflect"
)

func HighestTransactions(amounts []int) int {
	if len(amounts) == 0 { return 0 }
	m := amounts[0]
	for _, v := range amounts {
		if v > m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, HighestTransactions([]int{10, 20, 30})},
		{[]int{50, 60, 70}, HighestTransactions([]int{50, 60, 70})},
		{[]int{1, 2, 3}, HighestTransactions([]int{1, 2, 3})},
		{[]int{100, 200}, HighestTransactions([]int{100, 200})},
		{[]int{}, HighestTransactions([]int{})},
	}
	
	for idx, t := range tests {
		res := HighestTransactions(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think
